datafusion/sql/src/expr/mod.rs

```diff
@@ -695,6 +696,23 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         not_impl_err!("Struct not supported by ExprPlanner: {create_struct_args:?}")
     }
 
+    fn parse_tuple(
+        &self,
+        schema: &DFSchema,
+        planner_context: &mut PlannerContext,
+        values: Vec<SQLExpr>,
+    ) -> Result<Expr> {
+        match values.first() {
+            Some(SQLExpr::Identifier(_)) | Some(SQLExpr::Value(_)) => {
+                self.parse_struct(schema, planner_context, values, vec![])
+            }
+            None => not_impl_err!("Empty tuple not supported yet"),
+            _ => {
```

Comment:
I was trying to figure out how to write a negative test for this case, however anything I could try seems to work

DataFusion CLI v41.0.0
> create table foo as values (1);

```sql
> select * from foo where (column1) IN ((column1+1), (2));
+---------+
| column1 |
+---------+
+---------+
0 row(s) fetched.
Elapsed 0.008 seconds.

> select * from foo where (column1) IN ((column1+1), (2), (1));
+---------+
| column1 |
+---------+
| 1       |
+---------+
1 row(s) fetched.
Elapsed 0.010 seconds.
```

Reply:
```rust
query error DataFusion error: This feature is not implemented: Only identifiers and literals are supported in tuples
select a from values where (a, c) IN ((abs(a), 1));
```
Query success in duckdb
```
D select a from t where (a, b) in ((1, abs(a)));
┌────────┐
│   a    │
│ int32  │
├────────┤
│ 0 rows │
└────────┘
```

We could support `abs()` and other complex type in tuple in follow PR, but we should check not only `first` element of the values but all the values in this PR

Reply:
I will file a ticket to track the feature

Reply:
> ```rust
> query error DataFusion error: This feature is not implemented: Only identifiers and literals are supported in tuples
> select a from values where (a, c) IN ((abs(a), 1));
> ```
> 
> Query success in duckdb
> 
> ```
> D select a from t where (a, b) in ((1, abs(a)));
> ┌────────┐
> │   a    │
> │ int32  │
> ├────────┤
> │ 0 rows │
> └────────┘
> ```
> 
> We could support `abs()` and other complex type in tuple in follow PR, but we should check not only `first` element of the values but all the values in this PR

I am not sure about whether our evaulation support struct array like `{(struct(Expr1{column1}, 1)), struct (Expr2{column2}, 2))}`?